plugin/qlight/gateway.go

```diff
@@ -0,0 +1,33 @@
+package qlight
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ConsenSys/quorum-qlight-token-manager-plugin-sdk-go/proto"
+)
+
+type PluginGateway struct {
+	client proto.PluginQLightTokenRefresherClient
+}
+
+var _ PluginTokenManager = &PluginGateway{}
+
+func (p *PluginGateway) TokenRefresh(ctx context.Context, currentToken, psi string) (string, error) {
+	resp, err := p.client.TokenRefresh(ctx, &proto.TokenRefresh_Request{
+		CurrentToken: currentToken,
+		Psi:          psi,
+	})
+	if err != nil {
+		return "", fmt.Errorf("refresh token: %w", err)
+	}
+	return resp.Token, nil
+}
+
+func (p *PluginGateway) PluginTokenManager(ctx context.Context) (int32, error) {
+	resp, err := p.client.PluginQLightTokenManager(ctx, &proto.PluginQLightTokenManager_Request{})
```

Comment:
One potential concern with having the plugin return a statically configured refresh interval is that it couples the plugin configuration to the oauth2 server configuration.  For example, if the the oauth2 server admin decides to change the expiration period of generated tokens (say from 1hr to 30mins, perhaps due to a change in policy) then this will require action from the quorum admin to update the plugin config with a new `RefreshAnticipationInMillisecond` and reload the plugin, otherwise requests will start failing with expired token errors.

If we're happy with that then fine, but if we want to address that then an alternative could be for the plugin to use the expiration time returned in the JWT token (if there is one) instead of having the quorum admin provide the expected refresh period in config.

Reply:
it's an anticipation, that is the expiry time/delay - anticipation time/delay.
If anticipation is greater than expiry, it will end with a refresh at every request.
In which case no refresh would happen?

Reply:
sure - and if the anticipation is longer than the expiry the request would fail because the token would have expired correct? in that case does it trigger an refresh?

Reply:
the refresh is made before the request